### PR TITLE
resources in dagstermill

### PR DESCRIPTION
**Pull request by puma314**
This is a work-in-progress PR on resources in dagstermill.

There is now a test case with a very simple resource use-case and it passes! To make resources work, I've injected more code in the notebook that manually calls `__enter__` on the resources so that they are entered within the notebook. The only concern is that `__exit__` is never called, so I'm not sure if there are going to be a bunch of zombie resources around after execution.

(Also I know that only calling __enter__ is not exactly what happens with the pep specification of the with ... as ... statement, but wanted more high-level thoughts before implementing out the nitty-gritty).

Any feedback on this mechanism / suggestions would be much appreciated!

### CODE REVIEW

**Comment by schrockn** [comment]
https://github.com/dagster-io/dagster/pull/896

for discussion

File: python_modules/dagstermill/dagstermill/__init__.py [hunk 1]
```diff
@@ -60,6 +65,9 @@
 class DagstermillInNotebookExecutionContext(AbstractTransformExecutionContext):
     def __init__(self, pipeline_context):
         self._pipeline_context = pipeline_context
```
Comment (schrockn) [inline python_modules/dagstermill/dagstermill/__init__.py]: if you add a check.inst_param up here I believe you will catch the bug I mentioned at 158

File: python_modules/dagstermill/dagstermill/__init__.py [hunk 2]
```diff
@@ -124,18 +154,9 @@ def register_repository(self, repository_def):
...
-        # manage this manually (without an if block) in order to make this work.
-        # See https://github.com/dagster-io/dagster/issues/796
-        with yield_pipeline_execution_context(
-            pipeline_def,
-            {} if context_config is None else {'context': context_config},
-            RunConfig(run_id=''),
-        ) as pipeline_context:
-            self.context = DagstermillInNotebookExecutionContext(pipeline_context)
+        environment_dict = {} if context_config is None else {'context': context_config}
+        pipeline_context = yield_blank_pipeline_execution_context(environment_dict)
```
Comment (schrockn) [inline python_modules/dagstermill/dagstermill/__init__.py]: So i'm a bit confused as to how this is working

File: python_modules/dagster/dagster/core/execution.py [hunk 3]
```diff
@@ -563,23 +599,19 @@ def _create_resources(pipeline_def, context_def, environment, execution_context,
...
     # can potentially have many resources so we need to use this abstraction.
     with ExitStack() as stack:
-        for resource_name in context_def.resources.keys():
-            resource_obj_or_gen = get_resource_or_gen(
-                pipeline_def, context_def, resource_name, environment, run_id
-            )
-
-            resource_obj = stack.enter_context(as_ensured_single_gen(resource_obj_or_gen))
-
+        for resource_name, resource_gen in resource_gens.items():
```
Comment (schrockn) [inline python_modules/dagster/dagster/core/execution.py]: 👍🏻

**Comment by mgasner** [comment]
Picked up in https://dagster.phacility.com/D263

**Pull request closed without merging**